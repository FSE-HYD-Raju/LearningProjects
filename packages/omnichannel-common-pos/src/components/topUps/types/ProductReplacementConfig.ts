interface ProductReplacementConfig {
	individualId?: string;
	agreementId?: string;
	productId?: string;
	productOfferingId?: string;
	replaceType?: string;
	configuration?: object;
}
export { ProductReplacementConfig };
