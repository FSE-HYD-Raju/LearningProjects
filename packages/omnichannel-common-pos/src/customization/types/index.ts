import { CustomizationBootstrapType, GetStoreConfig } from "./CustomizationBootstrapType";
import { CustomizationClass } from "./CustomizationClass";
import { CustomizationFunction } from "./CustomizationFunction";
import { CustomizationPointsMapping } from "./CustomizationPointsMapping";

export {
	GetStoreConfig,
	CustomizationBootstrapType,
	CustomizationClass,
	CustomizationFunction,
	CustomizationPointsMapping
};
