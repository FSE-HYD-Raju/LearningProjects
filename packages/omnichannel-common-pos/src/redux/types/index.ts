import { ValidatePortInType } from "../../components/product/msisdn/MsisdnPortInSelector";
import { ConnectFluxReduxChannelType } from "./ConnectFluxReduxChannelType";
import { Disposers } from "./Disposers";
import { FluxAlt } from "./FluxAlt";
import { FluxReduxMap } from "./FluxReduxMap";
import { FluxStore } from "./FluxStore";
import { ReduxFluxActionMigration } from "./ReduxFluxActionMigration";
import { FluxWithActions } from "./fluxActions/FluxWithActions";

export {
	ConnectFluxReduxChannelType,
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration,
	FluxWithActions,
	ValidatePortInType
};

export * from "./utils/Omit";
export * from "./utils/HocResult";
export * from "./fluxActions";
export * from "../models/cms/cms.types";
export * from "../models/auth/auth.types";
export { CmsState } from "../models/cms/cms.reducers";
export { CmsAdminState } from "../models/cmsAdmin/cmsAdmin.reducers";
export * from "../models/auth/auth.types";
export * from "../models/service/service.types";
export * from "../models/support/support.types";
export * from "../models/comparison/comparison.types";
export * from "../models/consul/consul.types";
export * from "../models/currency/currency.types";
export * from "../models/eCare/changeSim/changeSim.types";
export * from "../models/delivery/delivery.types";
export * from "../models/digitalLife/digitalLife.types";
export * from "../models/eCare/activateSim/activateSim.types";
export * from "../models/eCare/suspension/suspension.types";
export * from "../models/feature/feature.types";
export * from "../models/lifecycle/lifecycle.types";
export * from "../models/organization/organization.types";
export * from "../models/productOfferingConfiguration/productOfferingConfiguration.types";
export * from "../models/schema/schema.types";
export * from "../models/location/location.types";
export * from "../models/eligibility/eligibility.types";
export * from "../models/productOfferings/productOfferings.types";
export * from "../models/user/user.types";
export * from "../models/basket/basket.types";
export * from "../models/customer/customer.types";
export * from "../models/sales/sales.types";
export * from "./SupportCases";
export * from "../models/versionInformation/versionInformation.types";
export * from "../models/msisdnSelection/msisdnSelection.types";
export * from "../models/portIn/portIn.types";
export * from "../models/marketingConsent/marketingConsent.types";
export * from "../models/category/category.types";
export * from "../models/b2cCheckout/b2cCheckout.types";
export { ProvinceAndCityState, Province, CityType } from "../models/provinceAndCity/provinceAndCity.types";

export * from "./routes/HasPath";
export * from "./routes/ApplicationRoute";
export * from "./HasSchema";
export * from "./HasFlux";
export * from "./HasId";
export * from "./HasPrices";
export * from "./BillingInterval";
export * from "./Cardinality";
export * from "./Characteristic";
export * from "./CharacteristicValue";
export * from "./CommercialEnrichments";
export * from "./Country";
export * from "./Links";
export * from "./FilterLabel";
export * from "./Gender";
export * from "./ImageType";
export * from "./Msisdn";
export * from "./MsisdnConfig";
export * from "./Price";
export * from "./SimplePrice";
export * from "./PriceRange";
export * from "./PriceType";
export * from "./PriceRangeBoundaries";
export * from "./ProductFilter";
export * from "./ProductOffering";
export * from "./ProductOfferingGroup";
export * from "./AlternateProductOffering";
export * from "./StockLevel";
export * from "./ProductTypes";
export * from "./ServiceUnit";
export * from "./Resource";
export * from "./SimCard";
export * from "./UnitOfMeasure";
export * from "./Allowance";
export * from "./ValidityPeriod";
export * from "./ValueRegulator";
export * from "./AutoCharacteristicConfigurationType";
export * from "./ChargingBalances";
export * from "./ChargingBalanceChanges";
export * from "./ProductsModifyConfig";
export * from "./Product";
export * from "./Agreement";
export * from "./ContactMediaType";
export * from "./EmailAddress";
export * from "./PhoneNumber";
export * from "./PostalAddress";
export * from "./Location";
export * from "./StreetAddressTokens";
export * from "./IssuingAuthority";
export * from "./Identification";
export * from "./CommonPerson";
export * from "./RelatedPerson";
export * from "./OrderLifeCycleStatus";
export * from "./Order";
export * from "./OrderCancel";
export * from "./OrderItem";
export * from "./OrderTotalsType";
export * from "./Person";
export * from "./MsisdnLifecycleStatus";
export * from "./Basket";
export * from "./payment/ContextualPayment";
export * from "./payment/ContextualPaymentMethod";
export * from "./payment/ContextualPaymentValidation";
export * from "./payment/CustomerPaymentMethod";
export * from "./payment/CustomerPaymentMethodType";
export * from "./payment/PaymentMethodsFiltersType";
export * from "./payment/CreditCard";
export * from "./payment/Role";
export * from "./BasketItem";
export * from "./SpecSubType";
export * from "./SpecType";
export * from "./TargetLifecycleStatus";
export * from "./BasketLifecycleStatus";
export * from "./BasketValidationAction";
export * from "./BasketValidationInformation";
export * from "./BasketProduct";
export * from "./PersonToPersonRelationship";
export * from "./CustomerCase";
export * from "./Category";
export * from "./SubCategory";
export * from "./ProductLifecycleStatus";
export * from "./UsageLimits";
export * from "./UsageCounters";
export * from "./Identifier";
export * from "./Notification";
export * from "./Document";
export * from "./BillingBalance";
export * from "./ShippingMethod";
export * from "./ChangeResourceInitialization";
export * from "./ChangeResource";
export * from "./Reason";
export * from "./MsisdnReservation";
export * from "./payment/PaymentStatus";
export * from "./payment/PaymentInvocation";
export * from "./Owner";
export * from "./CustomerInteraction";
export * from "./Initiator";
export * from "./InteractionDetails";
export * from "./AttachedDocument";
export * from "./Discount";
export * from "./Interval";
export * from "./BasketPaymentReceipt";
export * from "./PaginationData";
export * from "./PersonsOrderData";
export * from "./PersonsOrderState";
export * from "./OrderStatusFilter";
export * from "./SalesInfo";
export * from "./Shipment";
export * from "./AxiosResponse";
export * from "./CommercialEnrichmentName";
export * from "./UsageEvent";
export * from "./UsagePrice";
export * from "./NonMonetaryAmount";
export * from "./EventDetails";
export * from "./LifecycleChangeAction";
export * from "./Service";
export * from "./ServiceLifeCycleStatus";
export * from "./Specification";
export * from "./CustomerAccount";
export * from "./CustomerAccountLifecycleStatus";
export * from "./RecurringTopUpModelType";
export * from "../../components/priceComponent/model/PriceComponentPriceType";
export * from "./ErrorType";
export * from "./PostalAddressValidationErrorType";
export * from "./ProductModification";
export * from "./CallForwardingServiceModify";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./PortInDecision";
export * from "./ProductOfferingFilter";
export * from "./ProductOfferingFilterMatcher";
export * from "./DomainContext";
export * from "./ServiceModification";
export * from "./ResourceStocks";
export * from "./ResourceInventories";
export * from "./Inventory";
export * from "./SalesOrganizationRole";
export * from "./NominationCharacteristics";
export * from "./NominationPOCharacteristicsConfig";
export * from "./BasketItemsColumn";
export * from "./OrderItemsColumn";
export * from "./OrdersColumn";
export * from "./BasketsColumn";
export * from "./ConsumptionData";
export * from "./ProductsReplaceType";
export * from "./OmnichannelApiResponse";
export * from "./ProductConfiguration";
export * from "./TopUpCharacteristic";
export * from "./RecurringTopUpIntervalType";
export * from "./RecurringTopUpCharacteristicsValue";
export * from "./RecurringTopUpProductType";
export * from "./HasProduct";
export * from "./Store";
export * from "./QueryStatus";
export * from "./DeviceReservation";
export * from "./HasIntl";
export * from "./ProductUsageColumn";
export * from "./WorkforceAppointment";
