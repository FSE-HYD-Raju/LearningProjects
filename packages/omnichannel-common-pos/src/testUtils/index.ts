import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
