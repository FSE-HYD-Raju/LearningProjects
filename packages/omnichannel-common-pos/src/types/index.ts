import { ContextType, contextTypesValidationMap } from "./ContextType";

export {
	ContextType,
	contextTypesValidationMap
};
