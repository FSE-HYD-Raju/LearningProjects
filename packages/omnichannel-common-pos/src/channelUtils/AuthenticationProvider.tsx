import * as React from "react";
import setAxiosInterceptor from "./setAxiosInterceptor";
import AALoginCallbackContainer from "../components/auth/AALoginCallbackContainer";
import { contextTypesValidationMap, ContextType } from "../types";
import { RouteComponentProps } from "react-router";

interface AuthenticationProviderStatePropsOwnProps {
	flux: any;
	store: any;
	children: React.ReactNode;
	loaderComponent: React.ComponentType<any>;
}

interface AuthenticationProviderStateProps {
	isAnonymousAuthenticationEnabled: boolean;
	isAuthenticated: boolean;
	location: {
		pathname: string;
		search?: string;
	};
	securedRoutes: Array<string>;
}

interface AuthenticationProviderActionProps {
	login: (anonymous: boolean, loginHint?: string) => void;
}

type AuthenticationProviderProps = AuthenticationProviderStatePropsOwnProps & AuthenticationProviderStateProps
	& AuthenticationProviderActionProps & RouteComponentProps<any>;

class AuthenticationProvider extends React.Component<AuthenticationProviderProps> {
	static contextTypes: React.ValidationMap<ContextType> = contextTypesValidationMap;

	public static AALOGIN_CALLBACK: string = "/aalogin/callback";
	public static AALOGOUT_REDIRECT: string = "/logout/redirect";
	public static AALOGOUT_REDIRECT_PARAM: string = "uri";

	constructor(props: AuthenticationProviderProps, context: ContextType) {
		super(props, context);
		setAxiosInterceptor({
			flux: props.flux,
			store: props.store,
			intl: context.intl
		});

		this.state = {
			locales: []
		};
	}

	componentWillMount() {
		if (this.props.location.pathname === AuthenticationProvider.AALOGOUT_REDIRECT) {
			const params: URLSearchParams = new URLSearchParams(this.props.location.search);
			const redirectUrl = params.get(AuthenticationProvider.AALOGOUT_REDIRECT_PARAM);
			if (redirectUrl) {
				window.location.replace(redirectUrl);
			}
		}
	}

	componentDidMount() {
		this.loginIfNeeded();
	}

	// Needed for login after redirect for some secure path
	componentDidUpdate() {
		this.loginIfNeeded();
	}

	loginIfNeeded() {
		if (this.shouldPerformLogin() && !this.isLoginCallback()) {
			this.props.login(!this.isInSecureRoute(), this.getLoginHintParam());
		}
	}

	/**
	 * Login should be performed if:
	 * - user is not authenticated AND
	 * - user is in secured route OR anonymous authentication is enabled
	 */
	shouldPerformLogin = () => {
		return !this.props.isAuthenticated && (this.isInSecureRoute() || this.props.isAnonymousAuthenticationEnabled);
	};

	/**
	 * Application should be rendered if:
	 * - login isn't needed AND
	 * - we are not in a login callback
	 */
	shouldRenderApplication = () => {
		return !this.shouldPerformLogin() && !this.isLoginCallback();
	};

	isLoginCallback = () => {
		return this.props.location.pathname === AuthenticationProvider.AALOGIN_CALLBACK;
	};

	/**
	 * Checks if current path is defined to be secured in active configuration.
	 * Configuration should be a list of paths, such as ["/digilife"].
	 */
	isInSecureRoute = () => {
		return this.props.securedRoutes.some(path => {
			return this.props.location.pathname.includes(path);
		});
	};

	getLoginHintParam = (): string | undefined => {
		const queryParams =
			this.props.location.search &&
			this.props.location.search
				.replace("?", "")
				.split("&")
				.reduce((params: any, param: string) => {
					const kv = param.split("=");
					params[kv[0]] = kv[1];
					return params;
				}, {})  || {};
		return queryParams.login_hint;
	};

	shouldComponentUpdate(nextProps: AuthenticationProviderProps) {
		return this.props.location.pathname !== nextProps.location.pathname ||
			this.props.location.search !== nextProps.location.search ||
			this.props.isAuthenticated !== nextProps.isAuthenticated ||
			this.props.securedRoutes !== nextProps.securedRoutes;
	}

	render() {
		const LoaderComponent = this.props.loaderComponent;
		return this.shouldRenderApplication() ? (
			<>{this.props.children}</>
		) : (
			<AALoginCallbackContainer flux={this.props.flux}>
				<LoaderComponent />
			</AALoginCallbackContainer>
		);
	}
}

export {
	AuthenticationProviderStatePropsOwnProps,
	AuthenticationProviderStateProps,
	AuthenticationProviderActionProps,
	AuthenticationProviderProps,
};

export default AuthenticationProvider;
