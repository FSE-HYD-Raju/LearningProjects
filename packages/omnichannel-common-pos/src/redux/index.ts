"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
