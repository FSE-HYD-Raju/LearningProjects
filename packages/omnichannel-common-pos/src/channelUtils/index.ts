import RootContainer from "./RootContainer";

import clearSession from "./clearSession";
import clientIntl from "./clientIntl";
import FluxBootstrapper from "./FluxBootstrapper";
import RootProvider, { RootProviderProps, AppComponentProps } from "./RootProvider";
import IntlContainer from "./IntlContainer";
import FormattedMessage, { FormattedMessageDescriptor } from "./FormattedMessage";
import FormattedHTMLMessage, { FormattedHTMLMessageDescriptor }  from "./FormattedHTMLMessage";
import FormattedTime from "./FormattedTime";
import FormattedNumber from "./FormattedNumber";
import FormattedNumberItalian from "./FormattedNumberItalian";
import FormattedDate from "./FormattedDate";
import AuthenticationProvider from "./AuthenticationProvider";
import AuthenticationProviderContainer from "./AuthenticationProviderContainer";
import Time, { TimeType } from "./Time";
const FluxMock = require("./tests/utils/FluxMock");

export {
	default as setAxiosInterceptor,
	setRequestHeaders
} from "./setAxiosInterceptor";

export {
	Time,
	TimeType,
	AuthenticationProvider,
	AuthenticationProviderContainer,
	clientIntl,
	clearSession,
	RootContainer,
	FluxBootstrapper,
	IntlContainer,
	FormattedMessage,
	FormattedHTMLMessage,
	FormattedTime,
	FormattedNumber,
	FormattedNumberItalian,
	FormattedDate,
	RootProvider,
	RootProviderProps,
	AppComponentProps,
	FormattedMessageDescriptor,
	FormattedHTMLMessageDescriptor,
	FluxMock
};
