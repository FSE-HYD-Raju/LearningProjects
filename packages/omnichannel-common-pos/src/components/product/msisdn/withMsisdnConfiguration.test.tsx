import * as React from "react";
import { mountWithContext, shallowWithContext } from "../../../testUtils";

import {
	MsisdnConfigurationProps,
	withMsisdnConfiguration,
	MsisdnConfigurationProviderProps
} from "./withMsisdnConfiguration";
import { ProductOffering } from "../../../redux/types";
import { FluxMock } from "../../../channelUtils";

const minPO: ProductOffering = {
	id: "123",
	name: "Test PO",
	categories: [],
	featureCharacteristics: [],
	inputCharacteristics: {},
	instanceCharacteristics: {},
	prices: [],
	commercialEnrichments: [],
	productOfferingGroups: [],
	productOfferings: [],
	selected: false,
	relationships: {},
};

const minProps: MsisdnConfigurationProviderProps<MsisdnConfigurationProps> = {
	product: minPO,
	configurations: {},
};

const TestComponent: React.FC<MsisdnConfigurationProps> = (props: MsisdnConfigurationProps) => {
	return (<div id="testComponent"/>);
};

describe("MsisdnConfiguraionProvider", () => {
	it("succeeds at shallow mount with minimal props", () => {
		const WithWrapper = withMsisdnConfiguration<MsisdnConfigurationProps>(TestComponent);

		const wrapper = shallowWithContext(<WithWrapper {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at mount  with minimal props", () => {
		const WithWrapper = withMsisdnConfiguration<MsisdnConfigurationProps>(TestComponent);

		mountWithContext(<WithWrapper {...minProps} />);
	});
});
