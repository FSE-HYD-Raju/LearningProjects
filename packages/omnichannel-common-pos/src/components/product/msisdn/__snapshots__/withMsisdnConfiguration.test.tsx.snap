// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MsisdnConfiguraionProvider succeeds at shallow mount with minimal props 1`] = `
<TestComponent
  configurations={Object {}}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  msisdnConfig={
    Object {
      "mandatoryConfigurations": Array [],
    }
  }
  msisdnModalVisible={false}
  product={
    Object {
      "categories": Array [],
      "commercialEnrichments": Array [],
      "featureCharacteristics": Array [],
      "id": "123",
      "inputCharacteristics": Object {},
      "instanceCharacteristics": Object {},
      "name": "Test PO",
      "prices": Array [],
      "productOfferingGroups": Array [],
      "productOfferings": Array [],
      "relationships": Object {},
      "selected": false,
    }
  }
  productNeedsMsisdnConfiguration={false}
  toggleMsisdnModal={[Function]}
  userOpened={false}
/>
`;
